src/openforms/registrations/contrib/zgw_apis/tests/test_validators.py

```diff
@@ -103,3 +103,105 @@ def test_no_zgw_api_group_and_no_default(self):
+                            "lengte": "10",
+                            "kardinaliteit": "1",
+                            "waardenverzameling": [],
+                        },
+                        toelichting="",
+                        zaaktype="https://zaken.nl/api/v1/zaaktypen/1",
+                    ),
+                ],
+            },
+        )
+
+        data = {
+            "zgw_api_group": zgw_group.pk,
+            "property_mappings": [
+                {"component_key": "textField", "eigenschap": "a property name"}
+            ],
+        }
+        serializer = ZaakOptionsSerializer(data=data)
+        is_valid = serializer.is_valid()
+
+        self.assertTrue(is_valid)
+        self.assertNotIn("property_mappings", serializer.errors)
+
+    @requests_mock.Mocker()
+    def test_provided_variable_does_not_exist_in_specific_zaaktype(self, m):
+        zgw_group = ZGWApiGroupConfigFactory.create(
+            zrc_service__api_root="https://zaken.nl/api/v1/",
+            drc_service__api_root="https://documenten.nl/api/v1/",
+            ztc_service__api_root="https://catalogus.nl/api/v1/",
+        )
+        zgw_group.zaaktype = "https://zaken.nl/api/v1/zaaktypen/1"
+        zgw_group.save()
+        m.get(
+            "https://catalogus.nl/api/v1/eigenschappen?zaaktype=https%3A%2F%2Fzaken.nl%2Fapi%2Fv1%2Fzaaktypen%2F1",
+            status_code=200,
+            json={
+                "count": 1,
+                "next": None,
+                "previous": None,
+                "results": [
+                    generate_oas_component(
+                        "catalogi",
+                        "schemas/Eigenschap",
+                        url="https://test.openzaak.nl/catalogi/api/v1/eigenschappen/1",
+                        naam="a property name",
+                        definitie="a definition",
+                        specificatie={
+                            "groep": "",
+                            "formaat": "tekst",
+                            "lengte": "10",
+                            "kardinaliteit": "1",
+                            "waardenverzameling": [],
+                        },
+                        toelichting="",
+                        zaaktype="https://zaken.nl/api/v1/zaaktypen/1",
+                    ),
+                ],
+            },
+        )
+
+        data = {
+            "zgw_api_group": zgw_group.pk,
+            "property_mappings": [
+                {"component_key": "textField", "eigenschap": "wrong variable"}
+            ],
+        }
+        serializer = ZaakOptionsSerializer(data=data)
+        is_valid = serializer.is_valid()
+
+        self.assertFalse(is_valid)
+        self.assertIn("property_mappings", serializer.errors)
+        self.assertEqual(
+            "Could not find a property with the name 'wrong variable' related to the zaaktype.",
+            serializer.errors["property_mappings"][0],
+        )
```

Comment:
I would expect an error `serializer.errors["property_mappings"][0]["eigenschap"][0]`, specific to the eigenschap field.

Let's make an improvement ticket for that for the new UI/design and leave this as is now.